torch/distributed/_tensor/dispatch.py

```diff
@@ -288,10 +284,13 @@ def operator_dispatch(
             else output_sharding.output_spec
         )
         out_dts = []
-        for i, out in enumerate(target_schema.func_schema.arguments.out):
-            out_dt = cast(dtensor.DTensor, kwargs[out.name])
-            out_dt._spec = cast(DTensorSpec, output_specs[i])
-            out_dts.append(out_dt)
+        spec_idx = 0
+        for arg in target_schema.func_schema.arguments:
+            if arg.is_out:
+                out_dt = cast(dtensor.DTensor, kwargs[arg.name])
+                out_dt._spec = cast(DTensorSpec, output_specs[spec_idx])
+                out_dts.append(out_dt)
+                spec_idx += 1
         return tuple(out_dts) if len(out_dts) > 1 else out_dts[0]
```

Comment:
General comment: This `return` assumes `len(out_dts) >= 1` or else there will be an index out of bounds error when accessing `out_dts[0]`. I am guessing that this `len(out_dts) >= 1` invariant always holds, but I just wanted to point this out. Maybe you want an assert?

Reply:
I'll land this as is as I don't want to trigger another full CI run, the assert is added in the follow up PR https://github.com/pytorch/pytorch/pull/90241

Reply:
nvm, might still need to change sth, will update.